From: Simon McVittie <smcv@collabora.com>
Date: Mon, 4 Nov 2019 10:21:51 +0000
Subject: gdbus-server-auth test: Include gcredentialsprivate.h

Otherwise we'll never test the EXTERNAL-only mode, because that relies
on testing the private macros
G_CREDENTIALS_UNIX_CREDENTIALS_MESSAGE_SUPPORTED and
G_CREDENTIALS_SOCKET_GET_CREDENTIALS_SUPPORTED.

Fixes: 9f962ebe "Add a test for GDBusServer authentication"
Signed-off-by: Simon McVittie <smcv@collabora.com>
Applied-upstream: 2.63.2, commit:49eccfbe709e73932048ab88cf905798b708222a
---
 gio/tests/gdbus-server-auth.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/gio/tests/gdbus-server-auth.c b/gio/tests/gdbus-server-auth.c
index 160c18b..d7312ea 100644
--- a/gio/tests/gdbus-server-auth.c
+++ b/gio/tests/gdbus-server-auth.c
@@ -22,6 +22,9 @@
 #include <glib/gstdio.h>
 #include <gio/gio.h>
 
+/* For G_CREDENTIALS_*_SUPPORTED */
+#include <gio/gcredentialsprivate.h>
+
 #ifdef HAVE_DBUS1
 #include <dbus/dbus.h>
 #endif
@@ -441,6 +444,10 @@ do_test_server_auth (InteropFlags flags)
   g_test_skip ("Testing interop with libdbus not supported");
 #endif /* !HAVE_DBUS1 */
 
+  /* No practical effect, just to avoid -Wunused-label under some
+   * combinations of #ifdefs */
+  goto out;
+
 out:
   if (server != NULL)
     g_dbus_server_stop (server);
